src/SqlBinding/TriggerBinding/SqlTableWatcher.cs

```diff
@@ -0,0 +1,604 @@
+        {
+            var getPrimaryKeysQuery =
+                $"SELECT c.name, t.name\n" +
+                $"FROM sys.indexes i\n" +
+                $"INNER JOIN sys.index_columns ic ON i.object_id = ic.object_id AND i.index_id = ic.index_id\n" +
+                $"INNER JOIN sys.columns c ON ic.object_id = c.object_id AND c.column_id = ic.column_id\n" +
+                $"INNER JOIN sys.types t ON c.user_type_id = t.user_type_id\n" +
+                $"WHERE i.is_primary_key = 1 and i.object_id = OBJECT_ID(\'{_userTable}\');";
+
+            using (SqlConnection connection = SqlBindingUtilities.BuildConnection(_connectionStringSetting, _configuration))
+            {
+                using (var getPrimaryKeysCommand = new SqlCommand(getPrimaryKeysQuery, connection))
+                {
+                    await connection.OpenAsync();
+                    using (SqlDataReader reader = await getPrimaryKeysCommand.ExecuteReaderAsync())
+                    {
+                        while (await reader.ReadAsync())
+                        {
+                            _primaryKeys.Add(reader.GetString(0), reader.GetString(1));
+                        }
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Queries the change/worker tables to check for new changes on the user's table
+        /// </summary>
+        /// <returns></returns>
+        private async Task CheckForChangesAsync()
+        {
+            using (SqlConnection connection = SqlBindingUtilities.BuildConnection(_connectionStringSetting, _configuration))
+            {
+                await connection.OpenAsync();
+                // TODO: Set up locks on transaction
+                SqlTransaction transaction = connection.BeginTransaction(System.Data.IsolationLevel.RepeatableRead);
+
+                using (SqlCommand getChangesCommand = BuildCheckForChangesCommand(connection, transaction))
+                {
+                    using (SqlDataReader reader = await getChangesCommand.ExecuteReaderAsync())
+                    {
+                        List<string> cols = new List<string>();
+                        while (await reader.ReadAsync())
+                        {
+                            _rows.Add(SqlBindingUtilities.BuildDictionaryFromSqlRow(reader, cols));
+                        }
+                    }
+                }
+
+                foreach (var row in _rows)
+                {
+                    // Not great that we're doing a SqlCommand per row, should batch this
+                    using (SqlCommand acquireLeaseCommand = BuildAcquireLeaseOnRowCommand(row, connection, transaction))
+                    {
+                        await acquireLeaseCommand.ExecuteNonQueryAsync();
+                        // Necessary so that other commands can re-use the parameters in the _primaryKeyValuesOfRows map
+                        acquireLeaseCommand.Parameters.Clear();
```

Comment:
Is this comment and `Clear()` call still valid? I don't see a `_primaryKeyValuesOfRows` field.